types/reactstrap/lib/Breadcrumb.d.ts

```diff
@@ -1,13 +1,14 @@
 import * as React from 'react';
 import { CSSModule } from '../index';
 
-export type BreadcrumbProps<T = {}> = React.HTMLAttributes<HTMLElement> & {
+export interface BreadcrumbProps extends React.HTMLAttributes<HTMLElement> {
+  [key: string]: any;
   tag?: string;
   className?: string;
   listTag?: string;
   listClassName?: string;
   cssModule?: CSSModule;
-} & T;
+}
 
-declare class Breadcrumb<T = {[key: string]: any}> extends React.Component<BreadcrumbProps<T>> {}
+declare class Breadcrumb<T = {[key: string]: any}> extends React.Component<BreadcrumbProps> {}
```

Comment:
I was on the side of not having the strict type option from the beginning, so this change is good with me. The original PR was intended to split the gap between two other contributors.

That being said, would something like
```typescript
import * as React from 'react';
import { CSSModule } from '../index';

export interface BreadcrumbProps extends React.HTMLAttributes<HTMLElement> {
  tag?: string;
  className?: string;
  listTag?: string;
  listClassName?: string;
  cssModule?: CSSModule;
}

declare class Breadcrumb<T = {[key: string]: any}> extends React.Component<BreadcrumbProps & T> {}
export default Breadcrumb;
```
still have the performance hit? It still cuts down on the number of types since we could keep the `index.html` from this merge, and if someone wanted the strictness, they could use the generic.

I'm not sure how you got your timing numbers, otherwise I would test that change myself.

Reply:
Yes, it would still be slow. :crying_cat_face:  50% of the slowdown is `BreadcrumpProps & T`, wherever you happen to put the intersection. The other 30% is the re-exports in index.d.ts, to make the default strict.

Reply:
I didn't realize the intersections had such a performance impact. Thanks for the explanation!

Reply:
They're a multiplier for other types that are already slow. React is actually the main culprit here; each subclass of React.Component creates multiple thousands of types.

I was trying to get DT back on its feet with this PR since the crash was blocking React PRs. React needs a more considered look at performance soon, though.